src/test/java/com/fasterxml/jackson/databind/cfg/CacheProviderTest.java

```diff
@@ -205,10 +249,12 @@ public void testBuilderValueValidation() throws Exception
         DefaultCacheProvider.builder()
                 .maxDeserializerCacheSize(0)
                 .maxSerializerCacheSize(0)
+                .maxTypeFactoryCacheSize(0)
```

Comment:
One thing I wonder: I think that most cache implementations impose a minimum size, so `0` (for example) won't do what user might want -- that is, disabling of caching.

Then again, it is now possibly to add custom cache instances too which can be "no-op" (do nothing) so maybe this doesn't matter.

I also haven't tested out if backing `LRUMap` honors 0 size or not: JDK Maps do have minimum size but since 2.14 `LRUMap` uses integrated version of a more optimized caching library (name of which escapes me now).

Reply:
`LRUMap` *does* honor max size of 0 (a write occurs, but is immediately evicted). `LRUMap` is based off [ConcurrentLinkedHashMap](https://github.com/ben-manes/concurrentlinkedhashmap), which is a predecessor to [Caffeine](https://github.com/ben-manes/caffeine)

Caffeine, on the other hand, batches evictions, so the zero maximum size constraint can be temporarily violated.

In addition, Cache2k, Android LruCache, and Ehcache all throw IllegalArgumentException when trying to create a cache with a maximum size of zero.

Reply:
Thank you @iProdigy . Sounds like we are good then. So 0 size for 2.14+ Jackson implementation will not retain anything in cache, and while there is write overhead that should be negligible.

It'd be nice to test this invariant, but no further work should be necessary it sounds.